[DOCS] Combining important config settings into a single page

lockewritesdocs opened the pull request:
This PR combines all information for important configuration settings into a single page, plus updating some formatting and incorporating minor edits.

Closes #63758 

Backports:
* 7.x #63884 
* 7.10 #63883

Commit:
Combining important config settings into a single page.

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

jrodewig reviewed (approved):
LGTM. However, instead of doing an [Indiana Jones-style switch](https://youtu.be/0gU35Tgtlmg?t=23), it's best practice to set up redirects for any removed pages. Merging PRs that fail the docs CI often ends in tears. However, if you feel comfortable, this may be worth an exception.

Unfortunately, that would also involve creating new anchors for the former pages and swapping out any xrefs in the ES docs. If you want help with that, let me know. Thanks for taking care of this!

Commit:
Updating ids for two pages causing link errors and implementing redirects.

lockewritesdocs commented:
> LGTM. However, instead of doing an [Indiana Jones-style switch](https://youtu.be/0gU35Tgtlmg?t=23), it's best practice to set up redirects for any removed pages. Merging PRs that fail the docs CI often ends in tears. However, if you feel comfortable, this may be worth an exception.
> 
> Unfortunately, that would also involve creating new anchors for the former pages and swapping out any xrefs in the ES docs. If you want help with that, let me know. Thanks for taking care of this!

Thanks for the suggestion @jrodewig! You're totally right that redirects are the way to go. Less brittle. I implemented redirects in my last commit and updated the IDs linking from other pages. ✅
